### PR TITLE
[TTAHUB-333] Allow fixing of legacy imports

### PR DESCRIPTION
**Pull request by thewatermethod**
## Description of change
Creates a simple UI for updating the users associated with a legacy report

## How to test
Steps work if you have a prod style dataset (no transformed data)
- The Jira has test cases. Visit one of the reports listed there as an admin and you'll see a "edit report users" button.
- Clicking the button allows you access to a form that lets you enter or change existing email addresses.

Note that the Jira's info is a little wonky (close but not correct in a few cases) so you should manually look up the users to see the actual email addresses. I don't want to include the specifics in this PR description since it's public.

After you update users in the UI, you can verify in the DB that the correct information is populated and the report is filterable and appears on the CSV.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-333

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: frontend/src/Grid.scss [hunk 1]
```diff
@@ -11,7 +11,7 @@
         display: grid;   
         grid-template-columns: 1fr 3fr;
         grid-template-areas: 
-            "grid-nav grid-content"
+            "grid-nav grid-content";
```
Comment (thewatermethod) [inline frontend/src/Grid.scss:14]: Just added missing semicolon to this unrelated file

File: frontend/src/fetchers/Admin.js [hunk 2]
```diff
@@ -140,3 +140,8 @@ export const importTrainingReports = async (data) => {
   }
   return res.json();
 };
+
+export const updateLegacyUsers = async (id, data) => {
```
Comment (thewatermethod) [inline frontend/src/fetchers/Admin.js:144]: added fetcher, which handles what happens when an admin edits the legacy users

File: frontend/src/fetchers/__tests__/Admin.js [hunk 3]
```diff
@@ -160,6 +161,15 @@ describe('Admin', () => {
     });
   });
```
Comment (thewatermethod) [inline frontend/src/fetchers/__tests__/Admin.js:163]: test for fetcher above

File: frontend/src/pages/LegacyReport/__tests__/index.js [hunk 4]
```diff
@@ -49,6 +56,36 @@ describe('LegacyReport', () => {
     expect(alert).toHaveTextContent('Unable to load activity report');
   });
```
Comment (thewatermethod) [inline frontend/src/pages/LegacyReport/__tests__/index.js:58]: These tests assert that 
1) an admin sees the button
2) clicking the button toggles the form
3) clicking the save button sends a put request

There is also a test for if the put request bombs

File: src/routes/admin/index.js [hunk 5]
```diff
@@ -31,5 +32,6 @@ router.use('/redis', redisRouter);
 router.use('/national-center', nationalCenterRouter);
 router.use('/email', emailRouter);
 router.use('/training-reports', trainingReportRouter);
+router.use('/legacy-reports', legacyReportRouter);
```
Comment (thewatermethod) [inline src/routes/admin/index.js:36]: Add new route to admin namespace on backend

**Review by AdamAdHocTeam — COMMENTED** [review]
Should this be a two part message 1 part success 1 part warning?

image

**Comment by thewatermethod** [comment]
> Should this be a two part message 1 part success 1 part warning?
> 
> image

Just trying not to over think the admin screens

File: src/routes/admin/legacyReports.ts [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+            userId: {
+              [Op.not]: collaborator.id,
+            },
+          },
+        }));
+      }
+    }
+    if (managers) {
+      const approverIds = [];
+      const managerialTalent = managers.split(';');
```
Comment (AdamAdHocTeam) [inline src/routes/admin/legacyReports.ts:100]: lol

**Review by AdamAdHocTeam — APPROVED** [review]
Works well.

Might want to consider adding a warning alert section for when we cant validate the user.

Also do we want to put any constraints on the users we allowed to be selected. Right now it seems we just do a simple "does the user exist" without taking into consideration region and permissions.

File: src/routes/admin/legacyReports.ts [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+      ...imported,
+      createdBy,
+      modifiedBy,
+      manager: managers,
+    });
+
+    const promises = [];
+
+    if (createdBy) {
+      const creator = await User.findOne({
```
Comment (AdamAdHocTeam) [inline src/routes/admin/legacyReports.ts:58]: Do we want to consider checking region and permissions here?

Comment (AdamAdHocTeam) [reply src/routes/admin/legacyReports.ts:58]: For example, can this user even create reports in the region for this report?
